File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2025-08-01/examples-java/TuningOptionsListTableRecommendationsFilteredForAnalyzeTable.java

import com.azure.resourcemanager.postgresqlflexibleserver.models.RecommendationTypeParameterEnum;
import com.azure.resourcemanager.postgresqlflexibleserver.models.TuningOptionParameterEnum;

/**
 * Samples for TuningOptionsOperation ListRecommendations.
 */
public final class Main {
    /*
     * x-ms-original-file:
     * specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2025-08-01/examples/
     * TuningOptionsListTableRecommendationsFilteredForAnalyzeTable.json
     */
    /**
     * Sample code: List available table recommendations, filtered to exclusively get those of ANALYZE TABLE type.
     * 
     * @param manager Entry point to PostgreSqlManager.
     */
    public static void listAvailableTableRecommendationsFilteredToExclusivelyGetThoseOfANALYZETABLEType(
        com.azure.resourcemanager.postgresqlflexibleserver.PostgreSqlManager manager) {
        manager.tuningOptionsOperations().listRecommendations("exampleresourcegroup", "exampleserver",
            TuningOptionParameterEnum.TABLE, RecommendationTypeParameterEnum.ANALYZE_TABLE,
            com.azure.core.util.Context.NONE);
    }
}

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2026-01-01-preview/examples-js/TuningOptionsListTableRecommendationsFilteredForAnalyzeTable.js
const { PostgreSQLManagementFlexibleServerClient } = require("@azure/arm-postgresql-flexible");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to lists available object recommendations.
 *
 * @summary lists available object recommendations.
 * x-ms-original-file: 2026-01-01-preview/TuningOptionsListTableRecommendationsFilteredForAnalyzeTable.json
 */
async function listAvailableTableRecommendationsFilteredToExclusivelyGetThoseOfAnalyzeTableType() {
  const credential = new DefaultAzureCredential();
  const subscriptionId = "ffffffff-ffff-ffff-ffff-ffffffffffff";
  const client = new PostgreSQLManagementFlexibleServerClient(credential, subscriptionId);
  const resArray = new Array();
  for await (const item of client.tuningOptions.listRecommendations(
    "exampleresourcegroup",
    "exampleserver",
    "table",
    { recommendationType: "AnalyzeTable" },
  )) {
    resArray.push(item);
  }

  console.log(resArray);
}
